modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/programs/Download.scala

```diff
@@ -65,8 +67,23 @@ object Download {
         .flatMap { result =>
           val ((snapshot, context), observationLimit) = result
 
-          identifierStorage.get.flatMap { currencyAddress =>
-            consensus.manager.startFacilitatingAfterDownload(observationLimit, snapshot, CurrencySnapshotContext(currencyAddress, context))
+          def setCalculatedState = maybeDataApplication.map { da =>
+            implicit val d = da.calculatedStateDecoder
+
+            clusterStorage.getResponsivePeers
+              .map(NodeState.ready)
+              .map(_.toList)
+              .flatMap(Random[F].shuffleList)
+              .flatMap {
+                case Nil       => (new Throwable(s"No peers to fetch off-chain state from")).raiseError[F, DataCalculatedState]
+                case peer :: _ => p2pClient.dataApplication.getCalculatedState.run(peer)
```

Comment:
There is no way of telling whether this offChain state corresponds to the currency snapshot set here or if it's the previous/next one. Seems like this could be an issue. My intuition tells me that we should be able to tie the off-chain state to the snapshot e.g. by using ordinal.

Reply:
I think we should also store a hash of the off chain state in the currency snapshot so that when the state is downloaded it can be hashed and verified.

Reply:
done :+1: